workers/unity/Packages/io.improbable.gdk.core/Utility/EntityTemplate.cs

```diff
@@ -94,6 +131,17 @@ public void AddComponent<TSnapshot>(TSnapshot snapshot, string writeAccess)
             return null;
         }
 
+        /// <summary>
+        ///     Attempts to get a component snapshot stored in the EntityTemplate.
+        /// </summary>
+        /// <param name="componentId">The ID of the component to fetch.</param>
+        /// <returns>The component snapshot, if the component snapshot exists, null otherwise.</returns>
+        public ISpatialComponentSnapshot GetComponent(uint componentId)
```

Comment:
This is a little confusing, I'd rather make this a `TryGet` method.
`ISpatialComponentSnapshot` is always a struct otherwise, so having this return null feels odd

Reply:
I'm fine to add a `TryGet` as well. 

`ISpatialComponentSnapshot` could not be a struct, this function doesn't know that and we can't constrain it as such.

Reply:
Added `TryGetComponent` methods for both typed and type-erased variants